operators/nvidia_video_codec/nv_video_decoder/nv_video_decoder.cpp

```diff
@@ -0,0 +1,314 @@
+    }
+  }
+
+  auto allocator =
+      nvidia::gxf::Handle<nvidia::gxf::Allocator>::Create(context.context(), allocator_->gxf_cid());
+
+  auto output = nvidia::gxf::Entity::New(context.context());
+  if (!output) {
+    throw std::runtime_error("Failed to allocate message for output");
+  }
+
+  auto maybe_video_buffer = output.value().add<nvidia::gxf::VideoBuffer>();
+  if (!maybe_video_buffer) {
+    throw std::runtime_error("Failed to allocate video buffer");
+  }
+
+  auto video_buffer = maybe_video_buffer.value();
+  nvidia::gxf::VideoTypeTraits<nvidia::gxf::VideoFormat::GXF_VIDEO_FORMAT_NV12> video_type;
+  nvidia::gxf::VideoFormatSize<nvidia::gxf::VideoFormat::GXF_VIDEO_FORMAT_NV12> color_format;
+
+  int nFrame = 0;
+  uint8_t* pVideo = NULL;
+  int nVideoBytes = 0;
+  uint8_t* pFrame;
+  int nFrameReturned = 0;
+
+  do {
+    demuxer_->Demux(&pVideo, &nVideoBytes);
+
+    try {
+      nFrameReturned = decoder_->Decode(pVideo, nVideoBytes);
+    } catch (const std::exception& e) {
+      HOLOSCAN_LOG_ERROR("Decode failed: {}", e.what());
+      // Skip this frame and continue
+      continue;
+    }
+
+    if (nFrameReturned == 0) {
+      // emit_empty_frame(op_output, context);
+      continue;
+    }
+
+    nFrame += nFrameReturned;
+
+    // HOLOSCAN_LOG_INFO("Decoded {} frames, total {} frames", nFrameReturned, nFrame);
+    if (nFrameReturned == 0) {
+      continue;
+    }
+    if (nFrameReturned > 1) {
+      HOLOSCAN_LOG_WARN("More than one frame returned from decoder: {}. Processing all frames.",
+                        nFrameReturned);
+    }
+
+    for (int i = 0; i < nFrameReturned; i++) {
+      pFrame = decoder_->GetLockedFrame();
+
+      auto width = decoder_->GetWidth();
+      auto height = decoder_->GetHeight();
+      auto color_planes = color_format.getDefaultColorPlanes(width, height, false);
+      nvidia::gxf::VideoBufferInfo video_buffer_info{
+          static_cast<uint32_t>(width),
+          static_cast<uint32_t>(height),
+          video_type.value,
+          std::move(color_planes),
+          nvidia::gxf::SurfaceLayout::GXF_SURFACE_LAYOUT_PITCH_LINEAR};
+      video_buffer_info.color_planes[0].offset = 0;
+      video_buffer_info.color_planes[1].offset = decoder_->GetLumaPlaneSize();
+
+      auto result = video_buffer->resize<nvidia::gxf::VideoFormat::GXF_VIDEO_FORMAT_NV12>(
+          static_cast<uint32_t>(width),
+          static_cast<uint32_t>(height),
+          nvidia::gxf::SurfaceLayout::GXF_SURFACE_LAYOUT_PITCH_LINEAR,
+          nvidia::gxf::MemoryStorageType::kDevice,
+          allocator.value(),
+          false);
+
+      if (!result) {
+        throw std::runtime_error("Failed to resize video buffer");
+      }
+      size_t src_pitch = decoder_->GetDeviceFramePitch();
+      size_t dst_pitch = video_buffer_info.color_planes[0].stride;
+
+      // Log video buffer and decoder info for debugging
+      if (verbose_.get()) {
```

Comment:
Why not `HOLOSCAN_LOG_DEBUG`?

Reply:
HOLOSCAN_LOG_DEBUG produces too much noise and obscures the necessary information.

Reply:
Pushing back slightly: `HOLOSCAN_LOG_DEBUG` seems like the standard way for Holoscan apps to manage verbose information across many operators, as opposed to each app having a "verbose" setting that must be individually toggled on/off.

> too much noise and obscures the necessary information

Acknowledged, especially had this issue with previous GXF Multimedia extension logging. In the case of verbose debug output I'd expect that devs are typically debugging by reviewing a log file rather than watching the console directly. But there may not be a "one-size-fits-all" solution here.

Broadly, I wonder if @gigony has thoughts on best practices for a "verbose" flag in individual operators vs aligning to `DEBUG` logging in general Holoscan SDK apps+ops? Not a blocker for this PR.